packages/shared/lib/participation/staking.ts

```diff
@@ -464,3 +487,209 @@ export const hasAccountReachedMinimumAirdrop = (): boolean => {
 
     return overview.assemblyRewards > 0 || overview.shimmerRewards > 0
 }
+
+function isValidPeriodNumber(airdrop: StakingAirdrop, periodNumber: number): boolean {
+    if (!airdrop || periodNumber < 1) return false
+
+    switch (airdrop) {
+        case StakingAirdrop.Assembly:
+            return periodNumber <= LAST_ASSEMBLY_STAKING_PERIOD
+        case StakingAirdrop.Shimmer:
+            return periodNumber <= LAST_SHIMMER_STAKING_PERIOD
+        default:
+            return false
+    }
+}
+
+function getStakingResultUrl(airdrop: StakingAirdrop, periodNumber: number): string {
+    switch (airdrop) {
+        case StakingAirdrop.Assembly:
+            return ASSEMBLY_STAKING_RESULT_URLS[periodNumber - 1] ?? ''
+        case StakingAirdrop.Shimmer:
+            return SHIMMER_STAKING_RESULT_URLS[periodNumber - 1] ?? ''
+        default:
+            return ''
+    }
+}
+
+async function fetchStakingResult(airdrop: StakingAirdrop, periodNumber: number): Promise<StakingPeriodResult> {
+    const stakingResultUrl = getStakingResultUrl(airdrop, periodNumber)
+    const stakingResultResponse = await fetch(stakingResultUrl, getJsonRequestOptions())
+
+    return stakingResultResponse.json()
+}
+
+function getCurrentStakingRewards(): AccountStakingRewards[] {
```

Comment:
Using the word current here makes me think this function is for the current event that is happening, not the currently cached reward. Perhaps name this getAccountStakingRewards and avoid the word current.